File: src/main/java/com/yauhenl/neuro/World.java
package com.yauhenl.neuro;

import java.util.ArrayList;
import java.util.List;


class World {
    private final List<Bloop> bloops = new ArrayList<>();
    private final List<Food> foods = new ArrayList<>();

    private Environment env;

    World(Environment environment, Integer bloopsCount, Integer foodCount) {
        env = environment;
        for (int i = 0; i < bloopsCount; i++) {
            bloops.add(new Bloop(env, i));
        }
        for (int i = 0; i < foodCount; i++) {
            foods.add(new Food(env));
        }
    }

    void draw() {
        for (int i = 0; i < bloops.size(); i++) {
            Bloop it = bloops.get(i);
            it.update(foods, bloops);
            if (it.isDead()) {
                bloops.remove(it);
            }
        }
        foods.forEach(Food::display);
        if (env.random(1) < 0.05f) {
            foods.add(new Food(env));
        }
    }
}

File: src/main/kotlin/com/yauhenl/kotlin/neuro/World.kt
package com.yauhenl.kotlin.neuro

import java.util.ArrayList

class World internal constructor(private val env: Environment, bloopsCount: Int, foodCount: Int) {
    private val bloops = ArrayList<Bloop>()
    private val foods = ArrayList<Food>()

    init {
        for (i in 0..bloopsCount - 1) {
            bloops.add(Bloop(env, i))
        }
        for (i in 0..foodCount - 1) {
            foods.add(Food(env))
        }
    }

    internal fun draw() {
        for (i in bloops.indices) {
            val it = bloops[i]
            it.update(foods, bloops)
            if (it.isDead) {
                bloops.remove(it)
            }
        }
        foods.forEach { it.display() }
        if (env.random(1f) < 0.05f) {
            foods.add(Food(env))
        }
    }
}